lib/chef/knife/config_list.rb

```diff
@@ -91,36 +95,45 @@ def run
 
       private
 
-      def render_table(profiles, padding: 2)
-        # Replace the home dir in the client key path with ~.
+      def render_table(profiles, padding: 1)
+        rows = []
+        # Render the data to a 2D array that will be used for the table.
         profiles.each do |profile|
+          # Replace the home dir in the client key path with ~.
           profile[:client_key] = profile[:client_key].to_s.gsub(/^#{Regexp.escape(Dir.home)}/, "~") if profile[:client_key]
+          profile[:profile] = "#{profile[:active] ? "*" : " "}#{profile[:profile]}"
+          rows << profile.values_at(:profile, :client_name, :client_key, :server_url)
         end
-        # Render the data to a 2D array that will be used for the table.
-        table_data = [["", "Profile", "Client", "Key", "Server"]] + profiles.map do |profile|
-          [profile[:active] ? "*" : ""] + profile.values_at(:profile, :client_name, :client_key, :server_url).map(&:to_s)
-        end
-        # Compute column widths.
-        column_widths = Array.new(table_data.first.length) do |i|
-          table_data.map { |row| row[i].length + padding }.max
-        end
-        # Special case, the first col gets no padding (because indicator) and last
-        # get no padding because last.
-        column_widths[0] -= padding
-        column_widths[-1] -= padding
-        # Build the format string for each row.
-        format_string = column_widths.map { |w| "%-#{w}.#{w}s" }.join("")
-        format_string << "\n"
-        # Print the header row and a separator.
-        table = ui.color(format_string % table_data.first, :green)
-        table << "-" * column_widths.sum
-        table << "\n"
-        # Print the rest of the table.
-        table_data.drop(1).each do |row|
-          table << format_string % row
+
+        table = TTY::Table.new(header: TABLE_HEADER, rows: rows)
+
+        # Rotate the table to vertical if the screen width is less than table width.
+        if table.width > TTY::Screen.width
+          table.orientation = :vertical
+          table.rotate
+          # Add a new line after each profile record.
+          table.render do |renderer|
+            renderer.border do
+              separator ->(row) { (row + 1) % TABLE_HEADER.size == 0 }
+            end
+            # Remove the leading space added of the first column.
+            renderer.filter = Proc.new do |val, row_index, col_index|
+              if col_index == 1 || (row_index) % TABLE_HEADER.size == 0
+                val.strip
+              else
+                val
+              end
+            end
+          end
+        else
+          table.render do |renderer|
+            renderer.border do
+              mid   "-"
+              style :green
+            end
+            renderer.padding = [0, padding, 0, 0] # pad right with 2 characters
+          end
         end
-        # Trim the last newline because ui.output adds one.
-        table.chomp!
       end
 
     end
```

Comment:
I moved `lib/chef/knife/config_list_profiles.rb` over the top of this file and then updated it so that the diff between the two looks like this:

```


Petes-MBP:chef pete$ diff lib/chef/knife/config_list.rb lib/chef/knife/config_list_profiles.rb
22,23c22,23
<     class ConfigList < Knife
<       banner "knife config list (options)"
---
>     class ConfigListProfiles < Knife
>       banner "knife config list-profiles (options)"
44a45,46
>         Chef::Log.warn("knife config list-profiles has been deprecated in favor of knife config list. This will be removed in the major release version!")
>
```

Reply:
Thanks. I was going to ask msys to make that sync now that the refactor was merged